robot-command/src/main/java/org/obolibrary/robot/MeasureCommand.java

```diff
@@ -0,0 +1,147 @@
+        true,
+        "select which set of metrics you would like to compute: essential (default), extended, all.");
+    options = o;
+  }
+
+  /**
+   * Name of the command.
+   *
+   * @return name
+   */
+  public String getName() {
+    return "measure";
+  }
+
+  /**
+   * Brief description of the command.
+   *
+   * @return description
+   */
+  public String getDescription() {
+    return "compute the metrics of an ontology";
+  }
+
+  /**
+   * Command-line usage for the command.
+   *
+   * @return usage
+   */
+  // todo: --output?
+  public String getUsage() {
+    return "robot measure --input <file> --output <output>";
+  }
+
+  /**
+   * Command-line options for the command.
+   *
+   * @return options
+   */
+  public Options getOptions() {
+    return options;
+  }
+
+  /**
+   * Handle the command-line and file operations for the MeasureOperation.
+   *
+   * @param args strings to use as arguments
+   */
+  public void main(String[] args) {
+    try {
+      execute(null, args);
+    } catch (Exception e) {
+      CommandLineHelper.handleException(e);
+    }
+  }
+
+  /**
+   * Given an input state and command line arguments, compute metrics in ontology. The input
+   * ontology is not changed.
+   *
+   * @param state the state from the previous command, or null
+   * @param args the command-line arguments
+   * @return the unchanged state
+   * @throws Exception on any problem
+   */
+  public CommandState execute(CommandState state, String[] args) throws Exception {
+    CommandLine line = CommandLineHelper.getCommandLine(getUsage(), getOptions(), args);
+    if (line == null) {
+      return null;
+    }
+
+    IOHelper ioHelper = CommandLineHelper.getIOHelper(line);
+
+    String metricsType = CommandLineHelper.getDefaultValue(line, "metrics", "essential");
+    String format = CommandLineHelper.getOptionalValue(line, "format");
+    String output =
+        CommandLineHelper.getRequiredValue(line, "output", "an output file must be specified");
+
+    File output_file = new File(output);
+    if (format == null) {
+      String extension = FilenameUtils.getExtension(output_file.getName()).replace("yml", "yaml");
+      if (LEGAL_FORMATS.contains(extension)) {
+        format = extension;
+      } else {
+        format = "tsv";
+      }
+    }
+
+    state = CommandLineHelper.updateInputOntology(ioHelper, state, line);
+    OWLReasonerFactory reasonerFactory = CommandLineHelper.getReasonerFactory(line);
+
+    // Override default reasoner options with command-line options
+    Map<String, String> reasonerOptions = ReduceOperation.getDefaultOptions();
+    for (String option : reasonerOptions.keySet()) {
+      if (line.hasOption(option)) {
+        reasonerOptions.put(option, line.getOptionValue(option));
+      }
+    }
+
+    MeasureOperation.measure(
+        state.getOntology(),
+        reasonerFactory,
+        metricsType,
+        format,
+        output_file,
+        ioHelper.getPrefixes());
```

Comment:
@beckyjackson Somehow this does not seem to take into account prefixes I add with `-- prefix` to the command; I there must be something wrong with my command perhaps? I though the global properties are somehow handled elsewhere?

Reply:
Maybe it works, it must.. What could be wrong with this:

```
--prefix 'DDPHENOALT: http://purl.obolibrary.org/obo/ddpheno#'
```
For some reason this is not picked up. In any case. I am rambling. Forget about it! :P Sorry.

Reply:
It should be handled at line 109 where you use the `line` to get a new `IOHelper`.